```
ERROR: test_Conv2d_missing_argument (__main__.TestNN)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_nn.py", line 1359, in test_Conv2d_missing_argument
    self.assertRaises(RuntimeError, lambda: c(None))
  File "/opt/python/3.6.1/lib/python3.6/unittest/case.py", line 728, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/opt/python/3.6.1/lib/python3.6/unittest/case.py", line 177, in handle
    callable_obj(*args, **kwargs)
  File "test_nn.py", line 1359, in <lambda>
    self.assertRaises(RuntimeError, lambda: c(None))
  File "/home/travis/virtualenv/python3.6.1/lib/python3.6/site-packages/torch/nn/modules/module.py", line 206, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/travis/virtualenv/python3.6.1/lib/python3.6/site-packages/torch/nn/modules/conv.py", line 244, in forward
    self.padding, self.dilation, self.groups)
  File "/home/travis/virtualenv/python3.6.1/lib/python3.6/site-packages/torch/nn/functional.py", line 41, in conv2d
    if input.dim() != 4:
AttributeError: 'NoneType' object has no attribute 'dim'
```

--- a/test/test_nn.py
+++ b/test/test_nn.py
@@ -1366,7 +1366,7 @@ class TestNN(NNTestCase):
 
         for invalid_dims, module in zip(invalid_input_dims, modules):
             for dims in invalid_dims:
-                input = Variable(torch.Tensor(torch.Size((3, )*dims)))
+                input = Variable(torch.Tensor(torch.Size((3, ) * dims)))
                 self.assertRaises(ValueError, lambda: module(input))
 
     def test_ConvTranspose2d_output_size(self):

--- a/test/test_nn.py
+++ b/test/test_nn.py
@@ -1309,14 +1309,6 @@ class TestNN(NNTestCase):
         self.assertIn('buf', l.state_dict())
         self.assertIs(l.state_dict()['buf'], buf)
 
-    def test_Conv2d_raises_error_on_incorrect_input_size(self):
-        conv = nn.Conv2d(in_channels=3, out_channels=8, kernel_size=3)
-        input = Variable(torch.randn(3, 10, 20))
-        self.assertRaises(ValueError, lambda: conv(input))
-
-        input = Variable(torch.randn(3, 1, 1, 10, 20))
-        self.assertRaises(ValueError, lambda: conv(input))
-
     def test_Conv2d_inconsistent_types(self):
         inputs = Variable(torch.randn(4, 1, 7, 7).float())
         weights = Variable(torch.randn(1, 1, 3, 3).double())
@@ -1362,6 +1354,20 @@ class TestNN(NNTestCase):
             run_test(benchmark=True)
         finally:
             torch.backends.cudnn.benchmark = b
+
+    def test_conv_modules_raise_error_on_incorrect_input_size(self):
+        modules = [nn.Conv1d(3, 8, 3), nn.ConvTranspose1d(3, 8, 3),
+                   nn.Conv2d(3, 8, 3), nn.ConvTranspose2d(3, 8, 3),
+                   nn.Conv3d(3, 8, 3), nn.ConvTranspose3d(3, 8, 3)]
+
+        invalid_input_dims = [(2, 4), (2, 4),
+                              (3, 5), (3, 5),
+                              (4, 6), (4, 6)]
+
+        for invalid_dims, module in zip(invalid_input_dims, modules):
+            for dims in invalid_dims:
+                input = Variable(torch.Tensor(torch.Size((3, )*dims)))
+                self.assertRaises(ValueError, lambda: module(input))
 
     def test_ConvTranspose2d_output_size(self):
         m = nn.ConvTranspose2d(3, 4, 3, 3, 0, 2)

--- a/torch/nn/functional.py
+++ b/torch/nn/functional.py
@@ -69,6 +69,9 @@ def conv1d(input, weight, bias=None, stride=1, padding=0, dilation=1,
         >>> inputs = autograd.Variable(torch.randn(20, 16, 50))
         >>> F.conv1d(inputs, filters)
     """
+    if input is not None and input.dim() != 3:
+        raise ValueError("Expected 3D tensor as input, got {}D tensor instead.".format(input.dim()))
+
     f = ConvNd(_single(stride), _single(padding), _single(dilation), False,
                _single(0), groups, torch.backends.cudnn.benchmark, torch.backends.cudnn.enabled)
     return f(input, weight, bias)
@@ -98,6 +101,10 @@ def conv3d(input, weight, bias=None, stride=1, padding=0, dilation=1,
         >>> inputs = autograd.Variable(torch.randn(20, 16, 50, 10, 20))
         >>> F.conv3d(inputs, filters)
     """
+
+    if input is not None and input.dim() != 5:
+        raise ValueError("Expected 5D tensor as input, got {}D tensor instead.".format(input.dim()))
+
     f = ConvNd(_triple(stride), _triple(padding), _triple(dilation), False,
                _triple(0), groups, torch.backends.cudnn.benchmark, torch.backends.cudnn.enabled)
     return f(input, weight, bias)
@@ -105,6 +112,26 @@ def conv3d(input, weight, bias=None, stride=1, padding=0, dilation=1,
 
 def conv_transpose1d(input, weight, bias=None, stride=1, padding=0,
                      output_padding=0, groups=1, dilation=1):
+    """Applies a 1D transposed convolution operator over an input signal
+    composed of several input planes, sometimes also called "deconvolution".
+
+    See :class:`~torch.nn.ConvTranspose1d` for details and output shape.
+
+    Args:
+        input: input tensor of shape (minibatch x in_channels x iW)
+        weight: filters of shape (in_channels x out_channels x kW)
+        bias: optional bias of shape (out_channels)
+        stride: the stride of the convolving kernel. Default: 1
+        padding: implicit zero padding on the input. Default: 0
+        groups: split input into groups, in_channels should be divisible by
+          the number of groups
+        output_padding: A zero-padding of 0 <= padding < stride that should be
+          added to the output. Default: 0
+        dilation: the spacing between kernel elements. Default: 1
+    """
+    if input is not None and input.dim() != 3:
+        raise ValueError("Expected 3D tensor as input, got {}D tensor instead.".format(input.dim()))
+
     f = ConvNd(_single(stride), _single(padding), _single(dilation), True,
                _single(output_padding),
                groups, torch.backends.cudnn.benchmark, torch.backends.cudnn.enabled)
@@ -132,6 +159,10 @@ def conv_transpose2d(input, weight, bias=None, stride=1, padding=0,
           added to the output. Can be a single number or a tuple. Default: 0
         dilation: the spacing between kernel elements. Default: 1
     """
+
+    if input is not None and input.dim() != 4:
+        raise ValueError("Expected 4D tensor as input, got {}D tensor instead.".format(input.dim()))
+
     f = ConvNd(_pair(stride), _pair(padding), _pair(dilation), True,
                _pair(output_padding), groups, torch.backends.cudnn.benchmark, torch.backends.cudnn.enabled)
     return f(input, weight, bias)
@@ -158,6 +189,9 @@ def conv_transpose3d(input, weight, bias=None, stride=1, padding=0,
           the number of groups
         dilation: the spacing between kernel elements. Default: 1
     """
+    if input is not None and input.dim() != 5:
+        raise ValueError("Expected 5D tensor as input, got {}D tensor instead.".format(input.dim()))
+
     f = ConvNd(_triple(stride), _triple(padding), _triple(dilation), True,
                _triple(output_padding), groups, torch.backends.cudnn.benchmark, torch.backends.cudnn.enabled)
     return f(input, weight, bias)

--- a/torch/nn/functional.py
+++ b/torch/nn/functional.py
@@ -38,7 +38,7 @@ def conv2d(input, weight, bias=None, stride=1, padding=0, dilation=1,
         >>> inputs = autograd.Variable(torch.randn(1,4,5,5))
         >>> F.conv2d(inputs, filters, padding=1)
     """
-    if input.dim() != 4:
+    if input is not None and input.dim() != 4:
         raise ValueError("Expected 4D tensor as input, got {}D tensor instead.".format(input.dim()))
 
     f = ConvNd(_pair(stride), _pair(padding), _pair(dilation), False,

